src/app/auth/resetPassword/index.tsx

```diff
@@ -0,0 +1,250 @@
+  const [hasLowercase, setHasLowercase] = useState(false);
+  const [hasNumber, setHasNumber] = useState(false);
+  const [hasLength, setHasLength] = useState(false);
+  const [isDifferent, setIsDifferent] = useState(false);
+  const [isMatching, setIsMatching] = useState(false);
+
+  const checkPassword = (text: string) => {
+    setPassword(text);
+    if (text !== '') {
+      if (text !== text.toLowerCase()) {
+        setHasUppercase(true);
+      } else {
+        setHasUppercase(false);
+      }
+      if (text !== text.toUpperCase()) {
+        setHasLowercase(true);
+      } else {
+        setHasLowercase(false);
+      }
+      if (/[0-9]/.test(text)) {
+        setHasNumber(true);
+      } else {
+        setHasNumber(false);
+      }
+      if (text.length >= 8) {
+        setHasLength(true);
+      } else {
+        setHasLength(false);
+      }
+      //need to check that it is different from old password
+    }
+  };
+
+  const checkConfirmPassword = (text: string) => {
+    setConfirmPassword(text);
+    if (text !== '') {
+      if (text == password) {
+        setIsMatching(true);
+      } else {
+        setIsMatching(false);
+      }
+    }
+  };
+
+  const changePassword = async () => {
+    setLoading(true);
+    const { error } = await updateUser({ password });
+
+    if (error) {
+      Alert.alert('Updating password failed');
+    } else {
+      await signOut();
+      router.replace('/auth/login');
+    }
+
+    setLoading(false);
+  };
+
+  return (
+    <View style={styles.container}>
+      <View>
+        <Text style={[globalStyles.h1]}>Create a New Password</Text>
+        <View style={styles.newPassword}>
+          <UserStringInput
+            placeholder="Password"
+            label="New password"
+            placeholderTextColor={colors.darkGrey}
+            onChange={text => {
+              setPassword(text);
+              checkPassword(text);
+            }}
+            value={password}
+            attributes={{
+              textContentType: 'password',
+              secureTextEntry: passwordTextHidden,
+            }}
+          >
+            <RNEIcon
+              name={passwordTextHidden ? 'visibility-off' : 'visibility'}
+              type="material"
+              style={styles.icon}
+              onPress={() => setPasswordTextHidden(!passwordTextHidden)}
+            />
+          </UserStringInput>
+        </View>
+
+        {password !== '' && (
+          <View style={styles.passwordComplexity}>
+            <Icon type={hasUppercase ? 'green_check' : 'grey_dot'} />
+            <Text
+              style={[
+                globalStyles.errorMessage,
+                styles.passwordErrorText,
+                hasUppercase
+                  ? { color: colors.textGreen }
+                  : { color: colors.textGrey },
+              ]}
+            >
+              At least 1 uppercase letter
+            </Text>
+          </View>
+        )}
```

Comment:
I think it might be a good idea to make this a component because it is used so much. My bad on that, I should have initially put that in the sprint.

Reply:
You can also use this component in the signup page where it is also used